libraries/plugins/common_api/config_api.cpp

```diff
@@ -0,0 +1,167 @@
+
+config_api::configs_by_api_type config_api::_instances_by_api = config_api::configs_by_api_type();
+
+config_api::config_api(const std::string& api_name)
+    : _api_name(api_name)
+    , max_blockchain_history_depth(_max_blockchain_history_depth)
+    , max_blocks_history_depth(_max_blocks_history_depth)
+    , max_budgets_list_size(_max_budgets_list_size)
+    , max_discussions_list_size(_max_discussions_list_size)
+    , lookup_limit(_lookup_limit)
+    , tags_to_analize_count(_tags_to_analize_count)
+{
+}
+
+config_api::config_api(config_api& config)
+    : _api_name(config.api_name())
+    , _max_blockchain_history_depth(config._max_blockchain_history_depth)
+    , _max_blocks_history_depth(config._max_blocks_history_depth)
+    , _max_budgets_list_size(config._max_budgets_list_size)
+    , _max_discussions_list_size(config._max_discussions_list_size)
+    , _lookup_limit(config._lookup_limit)
+    , _tags_to_analize_count(config._tags_to_analize_count)
+    , max_blockchain_history_depth(_max_blockchain_history_depth)
+    , max_blocks_history_depth(_max_blocks_history_depth)
+    , max_budgets_list_size(_max_budgets_list_size)
+    , max_discussions_list_size(_max_discussions_list_size)
+    , lookup_limit(_lookup_limit)
+    , tags_to_analize_count(_tags_to_analize_count)
+{
+}
+
+#define STR(field) BOOST_PP_STRINGIZE(field)
+
+void config_api::get_program_options(boost::program_options::options_description& cli,
+                                     boost::program_options::options_description& cfg)
+{
+    namespace po = boost::program_options;
+    // clang-format off
+    cli.add_options()(get_option_name(STR(max_blockchain_history_depth)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(max_blockchain_history_depth)).c_str())
+                     (get_option_name(STR(max_blocks_history_depth)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(max_blocks_history_depth)).c_str())
+                     (get_option_name(STR(max_budgets_list_size)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(max_budgets_list_size)).c_str())
+                     (get_option_name(STR(max_discussions_list_size)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(max_discussions_list_size)).c_str())
+                     (get_option_name(STR(lookup_limit)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(lookup_limit)).c_str())
+                     (get_option_name(STR(tags_to_analize_count)).c_str(), po::value<uint32_t>(),
+                      get_option_description(STR(tags_to_analize_count)).c_str());
+    // clang-format on
+    cfg.add(cli);
+}
+
+void config_api::set_options(const boost::program_options::variables_map& options)
+{
+    config_api clean_config(api_name());
+
+    if (options.count(get_option_name(STR(max_blockchain_history_depth))))
```

Comment:
I would prefere a simple macro or something else instead of these repeated  if/else